docarray/array/storage/milvus/backend.py

```diff
@@ -0,0 +1,320 @@
+        'int': TypeMap(type=DataType.INT64, converter=_safe_cast_int),
+        'bool': TypeMap(type=DataType.BOOL, converter=bool),
+    }
+
+    def _init_storage(
+        self,
+        _docs: Optional['DocumentArraySourceType'] = None,
+        config: Optional[Union[MilvusConfig, Dict]] = None,
+        **kwargs,
+    ):
+        config = copy.deepcopy(config)
+        if not config:
+            raise ValueError('Empty config is not allowed for Milvus storage')
+        elif isinstance(config, dict):
+            config = dataclass_from_dict(MilvusConfig, config)
+
+        if config.collection_name is None:
+            id = uuid.uuid4().hex
+            config.collection_name = 'docarray__' + id
+        self._config = config
+        self._config.columns = self._normalize_columns(self._config.columns)
+
+        self._connection_alias = f'docarray_{config.host}_{config.port}'
+        connections.connect(
+            alias=self._connection_alias, host=config.host, port=config.port
+        )
+
+        self._collection = self._create_or_reuse_collection()
+        self._offset2id_collection = self._create_or_reuse_offset2id_collection()
+        self._build_index()
+        super()._init_storage()
+
+        # To align with Sqlite behavior; if `docs` is not `None` and table name
+        # is provided, :class:`DocumentArraySqlite` will clear the existing
+        # table and load the given `docs`
+        if _docs is None:
+            return
+        elif isinstance(_docs, Iterable):
+            self.clear()
+            self.extend(_docs)
+        else:
+            self.clear()
+            if isinstance(_docs, Document):
+                self.append(_docs)
+
+    def _create_or_reuse_collection(self):
+        if has_collection(self._config.collection_name, using=self._connection_alias):
+            return Collection(
+                self._config.collection_name, using=self._connection_alias
+            )
+
+        document_id = FieldSchema(
+            name='document_id', dtype=DataType.VARCHAR, max_length=1024, is_primary=True
+        )
+        embedding = FieldSchema(
+            name='embedding', dtype=DataType.FLOAT_VECTOR, dim=self._config.n_dim
+        )
+        serialized = FieldSchema(
+            name='serialized', dtype=DataType.VARCHAR, max_length=65_535
+        )
+
+        additional_columns = []
+        for col, coltype in self._config.columns.items():
+            mapped_type = self._map_type(coltype)
+            if mapped_type == DataType.VARCHAR:
+                field_ = FieldSchema(name=col, dtype=mapped_type, max_length=1024)
+            else:
+                field_ = FieldSchema(name=col, dtype=mapped_type)
+            additional_columns.append(field_)
+
+        schema = CollectionSchema(
+            fields=[document_id, embedding, serialized, *additional_columns],
+            description='DocumentArray collection',
+        )
+        return Collection(
```

Comment:
Does this constructor actually build the collection in the database ?

Reply:
yes, that's what it looks like from checking their code